utils/components.py

```diff
@@ -298,3 +307,359 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+            print(f"""
+            ###################################################### ERROR ######################################################
+            # The version {model_version} for model {model_name} is already registered. You cannot overwrite a model version. #
+            ###################################################################################################################
+            """)
+            sys.exit(1)
+    except StoreError as store:
+        # Do nothing, just to avoid this exception to return failure
+        print(f"""
+        ########### INFO ##############
+        # Model Registry is available #
+        ###############################
+        """)
+        sys.exit(0)
+    except Exception as e:
+        print(f"""
+        ############# ERROR ###############
+        # Model Registry is not available #
+        ###################################
+        """)
+        raise
+
+
+@dsl.component(base_image=RUNTIME_GENERIC_IMAGE)
+def test_training_operator():
+    import sys
+
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+
+    config.load_incluster_config()
+
+    with client.ApiClient() as api_client:
+        api_instance = client.CustomObjectsApi(api_client)
+        group = "kubeflow.org"
+        version = "v1"
+        plural = "pytorchjobs"
+
+        try:
+            api_response = api_instance.list_cluster_custom_object(
+                group, version, plural
+            )
+            print("""
+            ######################### INFO ###########################
+            # Kubeflow Training Operator PyTorchJob CRD is available #
+            ##########################################################
+            """)
+        except ApiException as e:
+            print("""
+            #################################################### ERROR ######################################################################
+            # Kubeflow Training Operator PyTorchJob CRD is unavailable. Ensure your OpenShift AI installation has Training Operator enabled #
+            #################################################################################################################################
+            """)
+            sys.exit(1)
+
+
+@dsl.component(base_image=RUNTIME_GENERIC_IMAGE)
+def test_oci_model(output_oci_model_uri: str, output_oci_registry_secret: str):
+    import base64
+    import json
+    import sys
+
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+
+    def url_match(sources, target):
+        import re
+        import urllib
+
+        for source in sources:
+            source_split = source.split(".")
+            target_split = target.split(".")
+            if len(source_split) != len(target_split):
+                return False
+            for idx, val in enumerate(source_split):
+                regex = source_split[idx].replace("*", ".*")
```

Comment:
I don't think this will work because `source_split[idx]` may have unescaped RegEx characters. You could use `regex.escape` but this might be a bit error prone since Kubernetes uses [filepath.Match](https://pkg.go.dev/path/filepath#Match).